spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/DecisionTreeTrainer.java

```diff
@@ -0,0 +1,259 @@
+        }
+        Preconditions.checkArgument(!features.equals(Schema.Type.NULL), "Field to classify must not be of type null");
+      }
+      Schema.Field prediction = inputSchema.getField(labelField);
+      if (labelField == null) {
+        throw new IllegalArgumentException(String.format("Prediction field %s does not exists in the input schema",
+                                                         labelField));
+      }
+      Schema predictionSchema = prediction.getSchema();
+      Schema.Type predictionFieldType = predictionSchema.isNullableSimple() ?
+        predictionSchema.getNonNullable().getType() : predictionSchema.getType();
+      if (predictionFieldType != Schema.Type.DOUBLE) {
+        throw new IllegalArgumentException(String.format("Prediction field must be of type Double, but was %s.",
+                                                         predictionFieldType));
+      }
+    }
+
+    private List<String> getFeatureList(Schema inputSchema) {
+      List<String> fields = new LinkedList<>();
+      List<Schema.Field> inputSchemaFields = inputSchema.getFields();
+      List<String> excludeFeatures = new ArrayList<>();
+
+      if (!Strings.isNullOrEmpty(featuresToInclude)) {
+        fields = Arrays.asList(featuresToInclude.split(","));
+        for (String field : fields) {
+          field = field.trim();
+          Schema.Field inputField = inputSchema.getField(field);
+          if (inputField == null) {
+            throw new IllegalArgumentException(String.format("Field %s does not exists in the input schema", field));
+          }
+        }
+      } else {
+        if (!Strings.isNullOrEmpty(featuresToExclude)) {
+          excludeFeatures = Arrays.asList(featuresToExclude.split(","));
+        }
+        for (Schema.Field field : inputSchemaFields) {
+          String fieldName = field.getName();
+          if (!fieldName.equals(labelField)) {
+            if (!excludeFeatures.isEmpty()) {
+              if (!excludeFeatures.contains(fieldName)) {
+                fields.add(fieldName);
+              }
+            } else {
+              fields.add(fieldName);
+            }
+          }
+        }
+      }
+      return fields;
+    }
+
+    private Map<Integer, Integer> getCategoricalFeatureInfo(Schema inputSchema) {
+      List<String> featureList = getFeatureList(inputSchema);
+      Map<Integer, Integer> outputFieldMappings = new HashMap<>();
+      if (!Strings.isNullOrEmpty(cardinalityMapping)) {
+        for (String field : Splitter.on(',').trimResults().split(cardinalityMapping)) {
+          String[] value = field.split(":");
+          if (value.length == 2) {
+            outputFieldMappings.put(featureList.indexOf(value[0]), Integer.parseInt(value[1]));
+          } else {
+            throw new IllegalArgumentException(
+              String.format("Either key or value is missing for 'Categorical Feature  Cardinality Mapping'. Please " +
```

Comment:
this can also happen if there are more than 2 values. Probably better to just have some generic message like:

```
"Invalid categorical feature mapping '%s'. Please make sure it is in the format 'feature':'cardinality'."
```